FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,49 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url, completion: { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case let .success((data, response)):
+				completion(FeedLoaderResultMapper.map(data, response))
+			case .failure:
+				completion(.failure(RemoteFeedLoader.Error.connectivity))
+			}
+		})
+	}
+}
+
+private struct FeedLoaderResultMapper {
+	static func map(_ data: Data, _ response: HTTPURLResponse) -> FeedLoader.Result {
+		if response.statusCode == 200, let root = try? JSONDecoder().decode(RemoteFeedLoadImagesRoot.self, from: data) {
+			return .success(root.images)
+		} else {
+			return .failure(RemoteFeedLoader.Error.invalidData)
+		}
+	}
+
+	private struct RemoteFeedLoadImagesRoot: Decodable {
+		let items: [RemoteFeedLoadImage]
+		var images: [FeedImage] {
+			items.map { $0.feedImage }
```

Comment:
Suggestion: another way of using the `map` function is with key paths: 
```suggestion
			items.map(\.feedImage)
```